v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+    "| **arguments**                      | \\[Optional\\] Arguments to be passed to inference script. Possible argument is '--forecast_quantiles' followed by quantile values. |"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.contrib.automl.pipeline.steps import AutoMLPipelineBuilder\n",
+    "from azureml.train.automl.runtime._many_models.many_models_parameters import (\n",
+    "    ManyModelsInferenceParameters,\n",
+    ")\n",
+    "\n",
+    "output_file_name = \"parallel_run_step.csv\"\n",
+    "\n",
+    "mm_parameters = ManyModelsInferenceParameters(\n",
+    "    partition_column_names=time_series_id_column_names,\n",
+    "    time_column_name=time_column_name,\n",
+    "    target_column_name=target_column_name,\n",
+    ")\n",
+    "\n",
+    "inference_steps = AutoMLPipelineBuilder.get_many_models_batch_inference_steps(\n",
+    "    experiment=experiment,\n",
+    "    inference_data=test_dataset,\n",
+    "    node_count=2,\n",
+    "    process_count_per_node=8,\n",
+    "    compute_target=compute_target,\n",
+    "    run_invocation_timeout=300,\n",
+    "    output_datastore=output_test_data_ds,\n",
+    "    train_run_id=training_run.id,\n",
+    "    train_experiment_name=training_run.experiment.name,\n",
+    "    inference_pipeline_parameters=mm_parameters,\n",
+    "    append_row_file_name=output_file_name,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.pipeline.core import Pipeline\n",
+    "\n",
+    "inference_pipeline = Pipeline(ws, steps=inference_steps)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "inference_run = experiment.submit(inference_pipeline)\n",
+    "inference_run.wait_for_completion(show_output=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6.2 Retreive the test set predictions from the many models\n",
+    "\n",
+    "The forecasting pipeline forecasts the orange juice quantity for a Store by Brand. The pipeline returns one file with the predictions for each store and outputs the result to the forecasting_output Blob container. The details of the blob container is listed in 'forecasting_output.txt' under Outputs+logs. \n",
+    "\n",
+    "The following code snippet:\n",
+    "1. Downloads the contents of the output folder that is passed in the parallel run step \n",
+    "2. Reads the output file that has the predictions as pandas dataframe \n",
+    "3. Displays the top 5 rows of the predictions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from azureml.pipeline.core import StepRun\n",
+    "# ! pip show azureml-pipeline-core"
```

Comment:
Let us move this part to the top and say:
Please make sure that the azureml-pipeline-core package is installed.
```python
try:
    from azureml.pipeline.core import StepRun
except ModuleNotFoundError:
    print("Please install azureml-pipeline-core with the command: \n! pip install azureml-pipeline-core")
```

Reply:
done